utils/DeployUtils.ps1

```diff
@@ -259,24 +287,111 @@ function ConfigureScubaGearModule {
         $ErrorActionPreference = $CurrentErrorActionPreference
     }
     catch {
-        Write-Warning "Warning: Cannot test module manifest:"
-        Write-Warning "Stacktrace:"
-        Write-Warning $_.ScriptStackTrace
-        Write-Warning "Exception:"
+        Write-Warning ">>> Warning: Cannot test module manifest:"
+        # Write-Warning ">>> Stacktrace:"
+        # Write-Warning $_.ScriptStackTrace
+        Write-Warning ">>> Exception:"
         Write-Warning $_.Exception
-        Write-Error "Failed to test module manifest"
-        return $False
```

Comment:
why was this return false removed?

Reply:
If the function throws an error, it doesn't return anything.  These functions previously failed to stopped if an error condition was reached, which is why the code was being published without signing.  The Write-Error ensures that stops when it needs to stop.

Reply:
`Write-Error` generates a non-terminating error (see https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/write-error?view=powershell-7.4#description). Shouldn't you use `throw` if you're intending the function to stop execution?

Reply:
However, Addam and I agree that the safest/smartest thing to do is throw an exception in each of the sub-functions and catch that exception at the highest level function and terminate with exit code one.  IOW, we don't want to rely on GitHub and Pester's error handling settings to do the right thing.

In another branch (#1194) I have already completely refactored DeployUtils.  I will add all the throws and the exit code logic there.  It doesn't make sense to add this logic to DeployUtils on this branch, since this code has already been rewritten.  So if you will, please accept this now, trusting that I will provide all the right code on that other branch.